src/Core/Geographic/Extent.js

```diff
@@ -14,513 +14,574 @@ const CARDINAL = {
+            _c._values[0], this.east,
+            this.south, _c._values[1]);
+
+        return [northWest, northEast, southWest, southEast];
     }
-};
-
-const center = new Coordinates('EPSG:4326', 0, 0, 0);
-/**
- * Subdivide equally an extent from its center to return four extents:
- * north-west, north-east, south-west and south-east.
- *
- * @returns {Extent[]} An array containing the four sections of the extent. The
- * order of the sections is [NW, NE, SW, SE].
- */
-Extent.prototype.subdivision = function subdivision() {
-    this.center(center);
-
-    const northWest = new Extent(this.crs(),
-        this.west(), center._values[0],
-        center._values[1], this.north());
-    const northEast = new Extent(this.crs(),
-        center._values[0], this.east(),
-        center._values[1], this.north());
-    const southWest = new Extent(this.crs(),
-        this.west(), center._values[0],
-        this.south(), center._values[1]);
-    const southEast = new Extent(this.crs(),
-        center._values[0], this.east(),
-        this.south(), center._values[1]);
-
-    return [northWest, northEast, southWest, southEast];
-};
-
-Extent.prototype.copyWithTransform = function copyWithTransform(t, s, toCopy) {
-    if (!toCopy.isTiledCrs()) {
-        this._crs = toCopy._crs;
-        this._values[0] = (toCopy._values[0] - t.x) / s.x;
-        this._values[1] = (toCopy._values[1] - t.x) / s.x;
-        if (this._values[0] > this._values[1]) {
-            const temp = this._values[0];
-            this._values[0] = this._values[1];
-            this._values[1] = temp;
-        }
-        this._values[2] = (toCopy._values[2] - t.y) / s.y;
-        this._values[3] = (toCopy._values[3] - t.y) / s.y;
-        if (this._values[2] > this._values[3]) {
-            const temp = this._values[2];
-            this._values[2] = this._values[3];
-            this._values[3] = temp;
+    /**
+     * Apply transform and copy this extent to input.
+     * @param {*} t translation transform
+     * @param {*} s scale transform
+     * @param {*} toCopy Extent to copy after transformation.
+     */
+    copyWithTransform(t, s, toCopy) {
+        if (!toCopy.isTiledCrs()) {
+            this._crs = toCopy._crs;
+            this._values[0] = (toCopy._values[0] - t.x) / s.x;
+            this._values[1] = (toCopy._values[1] - t.x) / s.x;
+            if (this._values[0] > this._values[1]) {
+                const temp = this._values[0];
+                this._values[0] = this._values[1];
+                this._values[1] = temp;
+            }
+            this._values[2] = (toCopy._values[2] - t.y) / s.y;
+            this._values[3] = (toCopy._values[3] - t.y) / s.y;
+            if (this._values[2] > this._values[3]) {
+                const temp = this._values[2];
+                this._values[2] = this._values[3];
+                this._values[3] = temp;
+            }
         }
     }
-};
+}
+
+_extent = new Extent('EPSG:4326', [0, 0, 0, 0]);
```

Comment:
Why not defining it at its declaration ?

Reply:
Because `Extent` isn't defined yet.

Reply:
Oh right